diff --git a/sio/csvio/writer.go b/sio/csvio/writer.go
index 79ffe3586..05aca139a 100644
--- a/sio/csvio/writer.go
+++ b/sio/csvio/writer.go
@@ -40,7 +40,7 @@ func NewWriter(w io.WriteCloser, opts WriterOpts) *Writer {
 		writer:    w,
 		encoder:   encoder,
 		flattener: expr.NewFlattener(super.NewContext()),
-		header:    !opts.NoHeader,
+		header:    false,
 		types:     make(map[int]struct{}),
 	}
 }
@@ -97,6 +97,9 @@ func (w *Writer) Write(rec super.Value) error {
 					s = strings.TrimSuffix(s, ".")
 				}
 			}
+		} else {
+			// NULL value - render as "NULL"
+			s = "NULL"
 		}
 		w.strings = append(w.strings, s)
 	}
